config/samples/pipeline_v1beta1_pipeline.yaml

```diff
@@ -7,46 +7,60 @@ spec:
-                sourceKey: 'wizzbang'
-          nextTasks: ['buildPush']
+            name: test-wizzbang-task
+          inputsourceBindings:
+            - name: repoUnderTest
+              sourceKey: wizzbang
         - name: 'buildPush'
           taskRef:
-              name: build-push-task
-          sourceBindings:
-              - inputName: 'workspace'
-                sourceKey: 'wizzbang'
-          artifactStoreBindings:
-              - storeName: 'registry'
-                storeKey: 'stagingRegistry'
+            name: build-push-task
+          inputSourceBindings:
+            - name: 'workspace'
+              sourceKey: 'wizzbang'
+              passedConstraint: [test-wizzbang-task]
+          outputSourceBindings:
+            - name: 'registry'
+              sourceKey: 'stagingRegistry'
           params:
-              - name: 'pathToDockerfile'
-                value: 'build/Dockerfile'
+            - name: 'pathToDockerfile'
+              type: 'build/Dockerfile'
         - name: 'integrationTest'
           taskRef:
-              name: integration-test-wizzbang-task
-          sourceBindings:
-              - inputName: 'repoUnderTest'
-                sourceKey: 'wizzbang'
-          paramBindings:
-              - inputName: testImage
-                taskName: buildPush
-                taskOutputName: builtImage
-        - name: 'deploy'
+            name: integration-test-wizzbang-task
+          inputSourceBindings:
+            - name: repoUnderTest
+              sourceKey: wizzbang
+              passedConstraint: [test]
+          outputSourceBindings:
+            - name: stagingRegistry
+              sourceKey: stagingRegistry
+              passedConstraint: [build-push-task]
+        - name: deploy
           taskRef:
               name: deploy-with-helm
           params:
-              - name: pathToHelmCharts
-                value: deploy/helm
-          sourceBindings:
-              - inputName: 'repoToDeploy'
-                sourceKey: 'wizzbang'
+            - name: pathToHelmCharts
+              type: deploy/helm
+          inputSourceBindings:
+            - name: repoToDeploy
+              sourceKey: wizzbang
+              passedConstraint: [integration-test-wizzbang-task]
           clusterBindings:
-              - clusterName: prod
-          paramBindings:
-              - inputName: image
-                taskName: buildPush
-                taskOutputName: builtImage
-          prevTasks: ['integrationTest']
+            - clusterName: prod
+          outputSourceBindings:
+            - name: stagingRegistry
+              sourceKey: stagingRegistry
+              passedConstraint: [build-push-task]
+    resources:
+    - name: wizzbang
+      resourceRef: 
+        name: wizzbang
+      type: github
+      url: github.com/wizzbangcorp/wizzbang
+      branch: master
+      commit: HEAD
+    - name: stagingRegistry
+      resourceRef: 
+        name: stagingRegistry
```

Comment:
i'm trying to understand the `resourceRef` in the context of the `resource` - should there  be a `stagingRef` CRD yaml that accompanies this to demonstrate what is in the CRD being referred to here?

I think it would be simpler to avoid having a separate CRD for resources specifically, but maybe there is some additional functionality that I'm missing!

Reply:
yeah, I had second thoughts about the resource ref also, I will consider removing it